test/CMakeLists.txt

```diff
@@ -156,6 +148,12 @@ foreach(file ${files})
         $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-deprecated;-Wno-float-equal>
         $<$<CXX_COMPILER_ID:GNU>:-Wno-deprecated-declarations>
     )
+    # Force to compile with highest warning level and treat warnings as errors
+    if(MSVC)
+      target_compile_options(${testcase} PRIVATE /W4 /WX)
+    else()
+      target_compile_options(${testcase} PRIVATE -Wall -Wextra -pedantic -Werror)
+    endif()
```

Comment:
Given the previous `target_compile_options` is using cmake generator expression syntax, might be good to add generator expression here too.

How about?

```
target_compile_options(${testcase} PRIVATE
$<$<CXX_COMPILER_ID:MSVC>:/W4 /WX>>
$<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -pedantic -Werror>>

Reply:
Hey, yeah probably a good idea. I will change that!